### PR TITLE
Air Pollution Metrics

### PR DESCRIPTION
**Pull request by MahdiRezaeiDev**
### Air Pollution Metrics Project: build a mobile web app
In this branch, I have made the following changes:

- Added `react` to the project.
- Added `redux` to the project.
- Added `redux-toolkit` to the project.
- Added project `pages`.
- Added project `components`.
- Added components `style`.
- Added project `routing`.
- Added project `store`.
- Added project `reducers & action creators`.
- Added project functionality and components test.

### CODE REVIEW

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @MahdiSohaily , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please make sure all your tests are passed.
image

- [x] you are also required to mock the connection to the Redux Store. Please kindly add those tests

- [x] Your UI looks really good but as part of the requirements, you are supposed to follow the design guideline. Please kindly adjust the design

**Your UI**
image

**Design to follow**
image

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MahdiRezaeiDev** [comment]
> 

The API is different in my case and we can not have that kind of navigating like for this API

File: src/redux/slice.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      newState.loading = false;
+      const arr = payload.list[0];
+      const components = Object.entries(arr.components);
+      newState.data = components;
+      return newState;
+    });
+  },
+});
+
+// export const { increment, decrement, incrementByAmount } = counterSlice.actions;
```
Comment (Nemwel-Boniface) [inline src/redux/slice.js]: - [ ] Kindly ensure that you remove all commented-out, un used/ un wanted pieces of code from your code base. It makes it look cluttered and messy.

File: src/__test__/components.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        <Router>
+          <Provider store={store}>
+            <Details />
+          </Provider>
+        </Router>
+      </React.StrictMode>,
+    );
+    expect(details).toMatchSnapshot();
+  });
+});
```
Comment (Nemwel-Boniface) [inline src/__test__/components.test.js:50]: - [ ] As requested of you by the previous reviewer and as it is in the project requirements, you are required to Create unit tests for pure functions in your Redux code. As I can currently see, none of this has been handled. Kindly write unit tests for your Redux methods.

image

File: src/components/Pages/Details.jsx [hunk 3]
```diff
@@ -7,6 +8,7 @@ import NavBar from '../props/NavBar';
 import Footer from '../props/Footer';
 import back from '../../imgs/back.png';
 import Loading from '../../imgs/loading.gif';
+import '../../styles/details.css';
 
 function Details() {
```
Comment (Nemwel-Boniface) [inline src/components/Pages/Details.jsx:13]: - [ ] As you have been requested of you by the previous reviewer, it is mandatory that your submitted design looks alike to the one provided in the Behance design. Tis included themes, colors, typography etc. This said, kindly ensure that your Details page design looks like the one that is provided. This should be a "Zebra crossing effect"

**This is how your design currently looks like:**

image


**This is the desired design**

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @MahdiSohaily    :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job addressing some of the changes that were requested by a previous reviewer :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Pages/Details.jsx (hunk 3 above)
Comment (MahdiRezaeiDev) [reply src/components/Pages/Details.jsx:13]: The design has been updated accordingly.

File: src/__test__/components.test.js (hunk 2 above)
Comment (MahdiRezaeiDev) [reply src/__test__/components.test.js:50]: Unit test has been written accordingly.

File: src/redux/slice.js (hunk 1 above)
Comment (MahdiRezaeiDev) [reply src/redux/slice.js]: File updated.

**Review by omarsalem7 — APPROVED** [review]
Hi @MahdiSohaily 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**